### PR TITLE
Minor fixes -- dont use assert

### PR DESCRIPTION
**Pull request by marvin2k**
fixes separated from #22

the change in the exception can be discussed: Only changing this exception but not changing all the others which use the same style...

### CODE REVIEW

**Comment by marvin2k** [comment]
removed "rework of exception", rebased to current master, removed even more asserts.

File: base/Time.hpp [hunk 1]
```diff
@@ -96,8 +95,9 @@ namespace base
                 case Microseconds:
                     sprintf(buffer,"%s:%06d", time, uSecs);
                     break;
-                default: 
-                    assert(-1);
+                default:
+                    throw std::runtime_error("base::Time::toString(): invalid "
```
Comment (doudou) [inline base/Time.hpp]: should probably use std::invalid_argument instead of runtime_error here

File: base/JointTransform.hpp [hunk 2]
```diff
@@ -33,13 +33,24 @@ struct JointTransformVector : public base::NamedVector<JointTransform>
      */
     void setRigidBodyStates( const base::samples::Joints& joints, std::vector<base::samples::RigidBodyState>& rbs ) const
     {
-	assert( !joints.names.empty() );
+        if (joints.names.empty()) {
+            std::stringstream ss;
+            ss << "base::JointTransformVector::setRigidBodyStates(): "
+                  "the vector 'joints.names()' is empty";
+            throw std::runtime_error(ss.str());
```
Comment (doudou) [inline base/JointTransform.hpp]: why using a stringstream ??? The string is a constant

Comment (marvin2k) [inline base/JointTransform.hpp]: Hm, the intention was to be able to add the current values of some variables into the exception. Will remove it.

**Comment by marvin2k** [comment]
removed unnecessary `std::stringstream`, used `std::invalid_argument` where appropriate.

**Pull request merged**